packages/buttons/src/styled/StyledIcon.ts

```diff
@@ -41,7 +41,7 @@ export const StyledIcon = styled(({ children, isRotated, theme, ...props }) =>
   'data-garden-version': PACKAGE_VERSION
 })<IStyledIconProps>`
   transform: ${props => props.isRotated && `rotate(${props.theme.rtl ? '-' : '+'}180deg)`};
-  transition: transform 0.25s ease-in-out, color 0.25s ease-in-out;
+  transition: transform 0.25s ease-in-out;
```

Comment:
This knocks out the intended color transition for the icons. The problem can be seen more clearly with `disable` toggling. See attached...

![transition](https://user-images.githubusercontent.com/143773/219464875-8e7b8402-8d09-413a-8a67-4d0fd260532e.gif)